src/lib/storage/run_length_segment/run_length_segment_iterable.hpp

```diff
@@ -70,11 +70,26 @@ class RunLengthSegmentIterable : public PointAccessibleSegmentIterable<RunLength
       }
     }
 
+    void decrement() {
+      --_current_position;
+
+      if (_current_position <= *(_end_position_it - 1)) {
```

Comment:
`_end_position_it - 1` will be an illegal operation if _end_position_it is the `begin` iterator

Reply:
Good catch! Fixed in https://github.com/hyrise/hyrise/pull/1517/commits/7a2cb883246c0a28f506b909d07e4d95f5ff045b